bridge/svix-bridge/src/runtime/mod.rs

```diff
@@ -1,45 +1,137 @@
 use anyhow::Result;
-use std::sync::Arc;
+use deadpool::unmanaged::Pool;
+use deno_ast::{MediaType, ParseParams, SourceTextInfo};
+use deno_runtime::deno_core::{
+    serde_v8,
+    v8::{self},
+    JsRuntime,
+};
 
-use deno_core::{anyhow, serde_v8, v8, JsRuntime};
 use svix_bridge_types::{JsObject, TransformerInput, TransformerOutput};
-use threadpool::ThreadPool;
-use tokio::sync::{oneshot, Mutex};
+use tokio::sync::oneshot;
 
-#[derive(Clone, Debug)]
-pub struct TpHandle(pub Arc<Mutex<ThreadPool>>);
+struct Executor {
+    tx: std::sync::mpsc::Sender<Job>,
+    _handle: std::thread::JoinHandle<()>,
+}
+
+impl Default for Executor {
+    fn default() -> Self {
+        let (tx, rx) = std::sync::mpsc::channel::<Job>();
+        let _handle = std::thread::spawn(move || {
+            let mut runtime = JsRuntime::new(Default::default());
+            for Job { input, script, cb } in rx {
+                let ret = run_script_inner(&mut runtime, input, script);
+                if cb.send(ret).is_err() {
+                    tracing::error!("failed to send script output to caller");
+                }
+            }
+        });
+        Self { tx, _handle }
+    }
+}
+
+type Callback = oneshot::Sender<Result<TransformerOutput>>;
 
-impl TpHandle {
+struct Job {
+    input: TransformerInput,
+    script: String,
+    cb: Callback,
+}
+
+impl Executor {
+    async fn execute(
+        &mut self,
+        input: TransformerInput,
+        script: String,
+    ) -> Result<TransformerOutput> {
+        let (tx, rx) = oneshot::channel();
+        self.tx.send(Job {
+            input,
+            script,
+            cb: tx,
+        })?;
+        rx.await?
+    }
+}
+
+#[derive(Clone)]
+pub struct JsPooler {
+    executors: Pool<Executor>,
+}
+
+impl JsPooler {
     pub fn new() -> Self {
-        Self(Arc::new(Mutex::new(ThreadPool::default())))
+        const POOL_SIZE: usize = 4;
```

Comment:
Is there any merit in making this configurable or dynamic based on number of cores or something?

Reply:
Yeah... I guess that'd be smart.

Reply:
This is now exposed as `transformation_worker_count` in the config.